src/core/key.service.ts

```
@@ -510,4 +567,187 @@
+      existingKeyPair.status = KeyStatus.INACTIVE;
+      existingKeyPair.metadata = {
+        ...existingKeyPair.metadata,
+        keyType: existingKeyPair.keyType,
+        cryptosuite: existingKeyPair.cryptosuite,
+        created: existingKeyPair.metadata?.created || new Date().toISOString(),
+        status: KeyStatus.INACTIVE,
+        rotatedAt: new Date().toISOString()
+      } as KeyMetadata;
+
+      // Save the updated metadata for the old key
+      await this.saveKeyPair(keyId, existingKeyPair);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)